docs/kbs_attestation_protocol.md

```diff
@@ -1,220 +1,418 @@
+   sending HTTP `GET` requests to resource specific endpoints. The KBS responds
+   with a `Response` payload that includes the requested resource data. The
+   resource data is protected by the ephemeral public key passed via the
+   previously sent `Attestation` payload. Within the valid time of the HTTP
+   Cookie generated by the KBS during step 2, the KBC can directly
+   request resources or services from KBS, without going through steps 2 and 3.
+
+# Payloads
+
+## `Request`
 
 The payload format of the request is as follows:
 
-(Note that the `/*...*/` comments are not valid in JSON, and must not be used in real message.)
-
 ```json
 {
-    "request": {
-        /* Attestation protocol version number used by KBC */
-        "version": "0.1.0",
-        /* Type of HW-TEE platforms where KBC is located, such as "tdx", "sev-snp", etc. */
-        "tee": "",
-        /* Access token to avoid multiple attempts triggered by consecutive requests. */
-        "token": "",
-        /* Reserved fields are used to support some special requests sent by HW-TEE. */
-        "extra-params": {}
-    }
+    /* Attestation protocol version number used by KBC */
+    "version": "0.1.0",
+    /*
+     * Type of HW-TEE platforms where KBC is located,
+     * e.g. "intel-tdx", "amd-sev-snp", etc.
+     */
+    "tee": "$tee",
+    /* Reserved fields to support some special requests sent by HW-TEE. */
+    "extra-params": {}
 }
 ```
+<sub>Note that the `/*...*/` comments are not valid in JSON, and must not be
+used in real message.</sub>
 
-- `protocol-version`
 
-The protocol version number supported by KBC. KBS needs to judge whether this KBC can communicate 
-normally according to this field.
+- `version`
+
+The protocol version number supported by KBC. KBS needs to judge whether this
+KBC can communicate normally according to this field.
 
 - `tee`
 
-Used to declare the type of HW-TEE platform where KBC is located, the valid values now is `tdx`, `sgx` and `sev-snp`.
+Used to declare the type of HW-TEE platform where KBC is located, the valid
+values are `intel-tdx`, `intel-sgx` and `amd-sev-snp`.
+
+- `extra-params`
+
+In the run-time attestation scenario (Intel TDX and SGX, AMD SEV-SNP), the
+`extra-params` field is not used, so is set to the empty string. However, for
+the attestation of some special HW-TEE platforms, this field may be used to
+transfer some specific information. For example, some attestations follow the
+Diffie–Hellman key exchange protocol to first build a secure channel and
+transfer secret messages (Such as AMD SEV(-ES) pre-attestation).
 
-- `token`
+## `Challenge`
 
-If other requests have been made before this request, in the previous response KBS would return a 
-token to KBC. There are two cases for the contents of this field: 
+If the KBC does not own any KBS generated HTTP Cookie, or if the Cookie validity
+has expired, the KBC is not allowed to directly request resources. It must go
```

Comment:
> the KBC is not allowed to directly request resources.

Is this worth clarifying the meaning of "allowed" here? Presumably if the KBC attempts to use an expired cookie, the KBS will return an error?

Reply:
Yes. Clarified now.